core/src/test/java/kafka/test/AdminApiIntegrationTest.java

```diff
@@ -0,0 +1,132 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@ExtendWith(value = ClusterTestExtensions.class)
+@Tag("integration")
+public class AdminApiIntegrationTest {
+
+    @ClusterTest(types = Type.KRAFT,
+            serverProperties = @ClusterConfigProperty(key = "metric.reporters",
+                    value = "kafka.test.AdminApiIntegrationTest$GetIdClientTelemetry"))
+    public void testClientInstanceId(ClusterInstance clusterInstance) {
+        Map<String, Object> configs = new HashMap<>();
+        configs.put("bootstrap.servers", clusterInstance.bootstrapServers());
+        configs.put(AdminClientConfig.ENABLE_METRICS_PUSH_CONFIG, "true");
+        try (Admin admin = Admin.create(configs)) {
+            assertNotNull(admin.clientInstanceId(Duration.ofSeconds(3)));
+        }
+    }
+
+
+    @ClusterTest(types = Type.KRAFT, brokers = 3)
+    public void testAddRaftVoter(ClusterInstance clusterInstance) {
+        Map<String, Object> configs = new HashMap<>();
+        configs.put(AdminClientConfig.ENABLE_METRICS_PUSH_CONFIG, "true");
+        String servers = clusterInstance.bootstrapServers();
+        configs.put("bootstrap.servers", servers);
+        try (Admin admin = Admin.create(configs)) {
+            assertThrows(ExecutionException.class, () -> {
+                admin.addRaftVoter(1, Uuid.randomUuid(), Collections.emptySet()).all().get();
+            });
+        }
+    }
+
+    @ClusterTest(types = Type.KRAFT, brokers = 3)
+    public void testRemoveRaftVoter(ClusterInstance clusterInstance) {
+        Map<String, Object> configs = new HashMap<>();
+        configs.put(AdminClientConfig.ENABLE_METRICS_PUSH_CONFIG, "true");
+        configs.put("bootstrap.servers", clusterInstance.bootstrapServers());
+        try (Admin admin = Admin.create(configs)) {
+            assertThrows(ExecutionException.class, () -> {
+                admin.removeRaftVoter(1, Uuid.randomUuid()).all().get();
+            });
+        }
+    }
+
+    @ClusterTest(types = Type.KRAFT)
+    public void testMetrics(ClusterInstance clusterInstance) {
+        Map<String, Object> configs = new HashMap<>();
+        configs.put(AdminClientConfig.ENABLE_METRICS_PUSH_CONFIG, "true");
+        configs.put("bootstrap.servers", clusterInstance.bootstrapServers());
+        try (Admin admin = Admin.create(configs)) {
+            admin.metrics().forEach((metricName, metric) -> {
+                assertSame(metricName, metric.metricName());
+            });
+        }
+    }
+
+
+    public static class GetIdClientTelemetry implements ClientTelemetry, MetricsReporter {
```

Comment:
We might need a comment here for why we need this and where it is used because at the moment the idea mark this as not used.

Reply:
This is the better place to have comments. @gongxuanzhang please ignore my above comment: https://github.com/apache/kafka/pull/16658#discussion_r1691146288